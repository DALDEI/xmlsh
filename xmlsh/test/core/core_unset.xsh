# Test local and unset of vars

a="global"
b="global"


function f()
{
  local n=$1
  echo $n a $a 
  echo $n b $b 
  local a;
  a=$*
  echo $n a $a
  unset a
  echo $n a $a
  echo $n b $b
  b="set in f()"
  echo $n b $b
  unset b
  echo $n b $b
  shift 
  if [ $# -gt 0 ] ; then 
    a="a from $n"
    echo calling f from $n : $*
    f $*
    echo retured f $*
    echo $n a $a
    echo $n b $b
   fi
 
   
}

echo a $a
echo b $b
f x y z
echo a $a
echo b $b

  

