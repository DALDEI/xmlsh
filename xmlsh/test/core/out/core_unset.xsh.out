a global
b global
x a global
x b global
x a x y z
x a
x b global
x b set in f()
x b
calling f from x : y z
y a a from x
y b
y a y z
y a
y b
y b set in f()
y b
calling f from y : z
z a a from y
z b
z a z
z a
z b
z b set in f()
z b
retured f z
y a a from y
y b
retured f y z
x a a from x
x b
a global
b
